lmdeploy/pytorch/engine/model_agent.py

```diff
@@ -812,12 +814,18 @@ def _broadcast_config(cache_config):
         config = model_config.hf_config
         torch_dtype = model_config.dtype
         device_map = None
+        model_init_kwargs = dict(torch_dtype=torch_dtype)
+        model_init_kwargs.update(model_config.init_kwargs)
+        # deal with AutoModel or Non-AutoModel
+        if model_config.auto_model_cls is AutoModelForCausalLM:
+            model_init_kwargs['trust_remote_code'] = trust_remote_code
+            from_config_func = AutoModelForCausalLM.from_config
+        else:
+            from_config_func = model_config.auto_model_cls._from_config
+
         with init_empty_weights():
-            model = AutoModelForCausalLM.from_config(
-                config,
-                torch_dtype=torch_dtype,
-                trust_remote_code=trust_remote_code,
-                **model_config.init_kwargs)
+            warnings.simplefilter('ignore')
```

Comment:
It is not a good idea to setup a global config here.

Reply:
fixed

Reply:
https://docs.python.org/3/library/warnings.html#temporarily-suppressing-warnings